doc/source/templates/01_batch_inference/start.ipynb

```diff
@@ -190,27 +206,99 @@
     "class PredictCallable:\n",
     "    def __init__(self):\n",
     "        # <Replace this with your own model initialization>\n",
+    "        import torch\n",
     "        from torchvision import models\n",
+    "        from torchvision.models import ResNet152_Weights\n",
     "\n",
-    "        self.model = models.resnet152(pretrained=True)\n",
+    "        self.model = models.resnet152(weights=ResNet152_Weights.DEFAULT)\n",
     "        self.model.eval()\n",
     "        self.device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
     "        self.model.to(self.device)\n",
     "\n",
     "    def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
     "        # <Replace this with your own model inference logic>\n",
+    "        import torch\n",
+    "\n",
     "        input_data = torch.as_tensor(batch[\"image\"], device=self.device)\n",
     "        with torch.inference_mode():\n",
     "            pred = self.model(input_data)\n",
     "        return {\"predicted_class_index\": pred.argmax(dim=1).detach().cpu().numpy()}\n"
    ]
   },
   {
+   "attachments": {},
    "cell_type": "markdown",
-   "id": "fda0c298",
+   "id": "2d63f352",
    "metadata": {},
    "source": [
-    "Now, perform batch prediction using Ray Data! Ray Data will perform model inference using `NUM_WORKERS` copies of the `PredictCallable` class you defined."
+    "## Run batch inference\n",
+    "\n",
+    "We'll first configure the number of workers and the resource requirements of each worker.\n",
+    "\n",
+    "These defaults will assume that your cluster has 4 GPUs available.\n",
+    "Be sure to stay within the resource constraints of your Ray Cluster if autoscaling is not enabled.\n",
+    "\n",
+    "`NUM_GPUS_PER_WORKER` can be a fractional amount! This will leverage Ray's fractional resource allocation, which means you can schedule multiple batch inference workers to use the same GPU, assuming that the models can all fit in GPU memory."
```

Comment:
we shouldn't recommend fractional GPUs for batch inference. It leads to worse performance since there are extra copies of the model.

Instead, to maximize GPU utilization, you should just increase the batch size.